src/puppetlabs/puppetdb/jdbc.clj

```diff
@@ -538,6 +538,30 @@
   [^HikariConfig config metrics-registry]
   (.setMetricRegistry config metrics-registry))
 
+(defn block-on-migration-mismatch
+  "Compare the migration the local PDB knows about against the current level
+  in the database. If the database version is greater than the local version
+  this indicates that a migration has been applied from a different PDB
+  before the local PDB has been upgraded. Raising an exception here will
+  cause all connection attempts from Hikari to fail until the local PDB has
+  been upgraded"
+  [max-migration]
+  (str/join
+   " "
+   ["BEGIN;"
+    "DO $$"
+    "DECLARE"
+    "migration_level INTEGER;"
+    "BEGIN"
+    "SELECT max(version) INTO migration_level FROM schema_migrations;"
```

Comment:
Need to have this query account for the case on startup of a fresh db where the schema_migrations table doesn't exist yet.